s3tar.go

```diff
@@ -252,6 +288,18 @@ func concatObjAndHeader(ctx context.Context, svc *s3.Client, objectList []*S3Obj
 	return results, nil
 }
 
+func fetchS3ObjectHead(ctx context.Context, svc *s3.Client, nextObject *S3Obj) *s3.HeadObjectOutput {
+	Debugf(ctx, "fetching head for %s/%s", *&nextObject.Bucket, *nextObject.Key)
+	head, err := svc.HeadObject(ctx, &s3.HeadObjectInput{
```

Comment:
So one thing to consider as well is that every call to HeadObject counts as a GET request. This will essentially double the pricing to build the object. For example if you're trying to create an in-memory tarball with 1,000 Amazon S3 objects, you will need 1,000 GET requests to pull the data, then 1,000 requests to fetch the Head.

Reply:
Agreed. I don't think this is avoidable. This is why I decided to add the `--preserve-posix-metadata` command line argument, since there's no need to do these checks unless you are using FSX or Storage Gateway. I could move the check for `opts.PreservePOSIXMetadata` into `fetchS3ObjectHead` to make this more clear if you prefer.